### PR TITLE
OHM-382: added alert/warning when exporting keystore

### PR DESCRIPTION
**Pull request by kaweesi**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -140,7 +140,7 @@
     <script src="bower_components/d3/d3.js"></script>
     <script src="bower_components/angular-bootstrap/ui-bootstrap-tpls.js"></script>
     <script src="bower_components/angular-taglist/js/angular-taglist-directive.js"></script>
-    <script src="bower_components/eve/eve.js"></script>
+    <script src="bower_components/eve-raphael/eve.js"></script>
```
Comment (kaweesi) [inline app/index.html]: this was auto edited, any explanation for this @rcrichton, @hnnesv !!!

File: app/scripts/controllers/exportImport.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ angular.module('openhimConsoleApp')
       $scope.showRecordOptions.ContactGroups = false;
       $scope.showRecordOptions.Keystore = false;
     };
-
```
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: these auto file formattings happened and i see no problems keeping them

File: app/scripts/controllers/exportImport.js [hunk 3]
```diff
@@ -126,6 +126,11 @@ angular.module('openhimConsoleApp')
       var exportData = angular.copy( $scope.selectedExports );
       var textFile = null;
 
+      // OHM-382: warn when exporting key
+      if(exportData.Keystore.length > 0) {
```
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: this is the real required change by this issue

File: app/scripts/controllers/exportImport.js [hunk 4]
```diff
@@ -126,6 +126,11 @@ angular.module('openhimConsoleApp')
...
+        alert("The key will be exported and should be protected");
```
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: i wonder what could be the easiest way to manage locations within this node js framework if we in the future translated openHIM into other languages!

File: app/scripts/controllers/exportImport.js [hunk 5]
```diff
@@ -126,6 +126,11 @@ angular.module('openhimConsoleApp')
       var exportData = angular.copy( $scope.selectedExports );
       var textFile = null;
 
+      // OHM-382: warn when exporting key
+      if(exportData !== undefined && exportData.Keystore !== undefined && exportData.Keystore.length > 0) {
```
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: You don't need the !== undefined in here.

File: app/scripts/controllers/exportImport.js [hunk 6]
```diff
@@ -126,6 +126,11 @@ angular.module('openhimConsoleApp')
...
+        Alerting.AlertAddMsg('bottom', 'warning', 'the key will be exported and should be protected');
```
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: Add this line above to reset existing alerts: Alerting.AlertReset();

Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: Also I think make it a top alert, this will involve fixing a bug in the view too. (in views/exportImport.html change line 17 to alert in alerts.top)

Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: Maybe make the description more detailed: 'Warning: The server's TLS private key will be exported and should be protected!'

File: app/scripts/controllers/exportImport.js (hunk 5 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: i fell into into some NPE here that's why i had to do this wrapping

File: app/scripts/controllers/exportImport.js (hunk 6 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: can you clarify this part; "this will involve fixing a bug in the view too. (in views/exportImport.html change line 17 to alert in alerts.top)"

File: .gitignore [hunk 7]
```diff
@@ -3,6 +3,7 @@ dist
 .tmp
 .sass-cache
 .idea/**
+.project
```
Comment (kaweesi) [inline .gitignore]: i am using eclipse IDE and so i have to ignore its .project directory

File: app/scripts/controllers/exportImport.js (hunk 6 above)
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: I see in the view there is no alerts.top section.  It is called alerts.role for some reason.  That is a bug

File: app/scripts/controllers/exportImport.js (hunk 5 above)
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: That is very strange, I don't think the !== undefined makes any difference though.

File: app/scripts/controllers/exportImport.js (hunk 6 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: still not clear unfortunately to me

File: app/scripts/controllers/exportImport.js (hunk 5 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: o yeah, but i experienced it, will remove it and test once again to trace its cause

File: app/scripts/controllers/exportImport.js (hunk 2 above)
Comment (bausmeier) [inline app/scripts/controllers/exportImport.js]: The only problem with autoformatting is that it makes the relevant changes harder to review since there are unrelated changes in the diff. If you are going to make formatting changes then they should be done in a separate commit so that they can be reviewed independently of the actual change.

File: app/scripts/controllers/exportImport.js (hunk 6 above)
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: It will become clear if you try to make it a 'top' alert.

File: .gitignore (hunk 7 above)
Comment (napiergit) [inline .gitignore]: I guess it's fine then

File: app/scripts/controllers/exportImport.js (hunk 6 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: o yeah, fixed at: ec05a74

File: app/scripts/controllers/exportImport.js [hunk 8]
```diff
@@ -126,6 +126,12 @@ angular.module('openhimConsoleApp')
       var exportData = angular.copy( $scope.selectedExports );
       var textFile = null;
 
+      // OHM-382: warn when exporting key
+      Alerting.AlertReset();
```
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: This can go in the if statement

File: app/scripts/controllers/exportImport.js [hunk 9]
```diff
@@ -126,6 +126,12 @@ angular.module('openhimConsoleApp')
...
+      if(exportData !== undefined && exportData.Keystore !== undefined && exportData.Keystore.length > 0) {
+    	Alerting.AlertAddMsg('top', 'warning', 'Warning: The server\'s TLS private key will be exported and should be protected!');
```
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: This indentation is not quite right

File: app/scripts/controllers/exportImport.js [hunk 10]
```diff
@@ -126,6 +126,12 @@ angular.module('openhimConsoleApp')
       var exportData = angular.copy( $scope.selectedExports );
       var textFile = null;
 
+      // OHM-382: warn when exporting key
+      Alerting.AlertReset();
+      if(exportData !== undefined && exportData.Keystore !== undefined && exportData.Keystore.length > 0) {
```
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: you can leave out the !== undefined

**Comment by napiergit** [comment]
Hi @kaweesi there are still some changes that have not been made

File: app/scripts/controllers/exportImport.js (hunk 10 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: do you mean both of the undefined checks!! or one of them?

Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: both

File: app/scripts/controllers/exportImport.js (hunk 8 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: if the user exported keystore inclusive, the alert would be posted but without reloading page if they re-eported excluding keystore i needed to remove the alert that's why i put it out!! do you still want it enclosed within if statement?

File: app/scripts/controllers/exportImport.js (hunk 9 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: let me address this after @napiergit confirms the rest of my reviews here so that i can have one commit instead to have this all resolved!

File: app/scripts/controllers/exportImport.js (hunk 10 above)
Comment (kaweesi) [inline app/scripts/controllers/exportImport.js]: o, i think i should have made it !== null, even this has no impact! let re-run my HIM instance to check if i fall into the issue again that made me include this check

File: app/scripts/controllers/exportImport.js (hunk 8 above)
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: Good point.  This is fine as is then.

File: app/scripts/controllers/exportImport.js (hunk 10 above)
Comment (napiergit) [inline app/scripts/controllers/exportImport.js]: I would do this:
if(exportData && exportData.Keystore && exportData.Keystore.length > 0)

**Comment by kaweesi** [comment]
all should be now resolved @napiergit !!!

File: app/scripts/controllers/exportImport.js [hunk 11]
```diff
@@ -126,6 +126,12 @@ angular.module('openhimConsoleApp')
       var exportData = angular.copy( $scope.selectedExports );
       var textFile = null;
 
+      // OHM-382: warn when exporting key
```
Comment (bausmeier) [inline app/scripts/controllers/exportImport.js]: There's no need for this comment. The related ticket can be found from the Git history if required.

**Comment by kaweesi** [comment]
thanks  @napiergit and @bausmeier  for the reviews, i have addressed all your comments and squashed all my commits into one 👍
